Source/WebCore/page/EventHandler.cpp

```diff
@@ -4339,28 +4339,37 @@ void EventHandler::defaultBackspaceEventHandler(KeyboardEvent& event)
 
 void EventHandler::stopKeyboardScrolling()
 {
-    Ref protectedFrame = m_frame;
-    auto* view = m_frame.view();
-    if (!view)
+    auto* node = m_mousePressNode.get();
```

Comment:
Can we factor "mouse press node -> enclosing scrollable area" out into a function here somewhere? Because I think you are adding the second and third (at least?) copy of these.

Also, do the other clients /just/ look at mousePressNode? I would expect to be able to move the focus with the keyboard too, so I'm a bit surprised. What does the existing code do?

Reply:
> Can we factor "mouse press node -> enclosing scrollable area" out into a function here somewhere? Because I think you are adding the second and third (at least?) copy of these.

I'm having trouble finding where else in EventHandler.cpp there is similar code that can be factored out.


> Also, do the other clients /just/ look at mousePressNode? I would expect to be able to move the focus with the keyboard too, so I'm a bit surprised. What does the existing code do?

I see that from the non-smooth code, EventHandler::scrollOverflow first tries to get the node from m_frame.document()->focusedElement(), and then if that fails tries m_mousePressNode.get(). I _think_ this would address that issue, so I included that change in the new patch, but I'm not certain.